rails/app/javascript/components/Map.jsx

```diff
@@ -320,9 +350,12 @@ export default class Map extends Component {
 
   addHomeButton() {
     const homeButton = this.createHomeButton();
-    const navControl = document.getElementsByClassName(
-      "mapboxgl-ctrl-zoom-in"
-    )[0];
+    let navControl;
+    if (this.state.isMapLibre) {
+      navControl = document.getElementsByClassName("maplibregl-ctrl-zoom-in")[0];
+    } else {
+      navControl = document.getElementsByClassName("mapboxgl-ctrl-zoom-in")[0];
+    }
```

Comment:
I think we can use `querySelector` or `querySelectorAll` to the same effect, while using pseudo regex matchers so we can avoid the conditional:

```suggestion
    navControl = document.querySelectorAll("button[class$="-ctrl-zoom-in]")[0];
```